src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -390,10 +446,10 @@ private Task ProcessEventAsync(CompilationEvent e, CancellationToken cancellatio
 
         private async Task ProcessSymbolDeclaredAsync(SymbolDeclaredCompilationEvent symbolEvent, CancellationToken cancellationToken)
         {
-            // Create a task per-analyzer to execute analyzer actions.
-            // We execute analyzers in parallel, but for a given analyzer we execute actions sequentially.
-            var tasksMap = PooledDictionary<DiagnosticAnalyzer, Task>.GetInstance();
-
+            // Collect all the analyzer action executors grouped by analyzer.
+            // NOTE: Right now we execute all the actions sequentially, but there is scope to fine tune this to execute certain actions in parallel.
+            var actionsMap = PooledDictionary<DiagnosticAnalyzer, ArrayBuilder<Action>>.GetInstance();
```

Comment:
@mavasani we should avoid the terminology "thread safe".  It is an ambiguous term that doesn't give sufficient guidance to analyzer authors about the situation we are enabling.  Instead we should pick terminology that describes the way in which we plan to use such analyzers.